### PR TITLE
Add a user option to specify the migrated file extension

### PR DESCRIPTION
**Pull request by the-slow-one**
The `dpct` options is `--sycl-file-extension` with choices `dp-cpp` (default) `sycl-cpp` and `cpp`.

Resolves https://jira.devtools.intel.com/browse/CTST-5644

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Comment by zhimingwang36** [comment]
Please address the failed cases.  like:  dpct/autocomplete.c, clang format.  Then rerun the pre-ci.  Thanks.

File: clang/lib/DPCT/SaveNewFiles.cpp [hunk 1]
```diff
@@ -157,11 +157,15 @@ void rewriteFileName(clang::tooling::UnifiedPath &FileName,
...
       DpctGlobalInfo::getChangeExtensions().count(Extension.str())) {
     if (FileType & SPT_CudaSource)
-      path::replace_extension(CanonicalPathStr, "dp.cpp");
+      path::replace_extension(CanonicalPathStr,
+                              DpctGlobalInfo::getSYCLSourceExtension());
     else if (FileType & SPT_CppSource)
-      path::replace_extension(CanonicalPathStr, Extension + ".dp.cpp");
+      path::replace_extension(CanonicalPathStr,
+                              Extension +
+                                DpctGlobalInfo::getSYCLSourceExtension());
```
Comment (the-slow-one) [inline clang/lib/DPCT/SaveNewFiles.cpp]:
```suggestion
                                  DpctGlobalInfo::getSYCLSourceExtension());
```

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**